src/scala/codeGeneration.js

```diff
@@ -0,0 +1,373 @@
+import {
+  GraphQLError,
+  getNamedType,
+  isCompositeType,
+  isAbstractType,
+  isEqualType,
+  GraphQLScalarType,
+  GraphQLEnumType,
+  GraphQLList,
+  GraphQLNonNull,
+  GraphQLID,
+  GraphQLInputObjectType
+} from 'graphql'
+
+import  { isTypeProperSuperTypeOf } from '../utilities/graphql'
+
+import {
+  join,
+  wrap,
+} from '../utilities/printing';
+
+import {
+  packageDeclaration,
+  objectDeclaration,
+  caseClassDeclaration,
+  propertyDeclaration,
+  propertyDeclarations,
+  escapeIdentifierIfNeeded,
+  comment
+} from './language';
+
+import {
+  caseClassNameForPropertyName,
+  caseClassNameForFragmentName,
+  caseClassNameForInlineFragment,
+  operationClassName,
+  enumCaseName,
+  propertiesFromSelectionSet,
+  propertyFromField,
+  propertyFromInlineFragment,
+  propertyFromFragmentSpread
+} from './naming';
+
+import {
+  escapedString,
+  multilineString,
+  dictionaryLiteralForFieldArguments,
+} from './values';
+
+import {
+  possibleTypesForType,
+  typeNameFromGraphQLType,
+} from './types';
+
+import CodeGenerator from '../utilities/CodeGenerator';
+
+export function generateSource(context, options) {
+  const generator = new CodeGenerator(context);
+
+  generator.printOnNewline('//  This file was automatically generated and should not be edited.');
+  generator.printNewline();
+
+  if (context.namespace) {
+    packageDeclaration(generator, context.namespace);
+  }
+
+  context.typesUsed.forEach(type => {
+    typeDeclarationForGraphQLType(generator, type);
+  });
+
+  Object.values(context.operations).forEach(operation => {
+    classDeclarationForOperation(generator, operation);
+  });
+
+  Object.values(context.fragments).forEach(fragment => {
+    caseClassDeclarationForFragment(generator, fragment);
+  });
+
+  return generator.output;
+}
+
+export function classDeclarationForOperation(
+  generator,
+  {
+    operationName,
+    operationType,
+    rootType,
+    variables,
+    fields,
+    inlineFragments,
+    fragmentSpreads,
+    fragmentsReferenced,
+    source,
+    sourceWithFragments,
+    operationId
+  }
+) {
+  let objectName;
+  let protocol;
+
+  switch (operationType) {
+    case 'query':
+      objectName = `${operationClassName(operationName)}Query`;
+      protocol = 'me.shadaj.apollo.GraphQLQuery';
```

Comment:
What does the `me.shadaj.apollo` namespace refer to? Is this a runtime dependency on code you have written? Is there any way we can make this `com.apollographql`? :)